src/renderer/viz/expressions/Scale.js

```diff
@@ -0,0 +1,51 @@
+import BaseExpression from './base';
+import { mul, pow, div } from '../expressions';
+import { implicitCast, checkType } from './utils';
+
+/**
+ * Scale a width value to keep feature width constant in real space (meters).
+ * This will change the width in pixels at different zoom levels to enforce the previous condition.
+ *
+ * @param {Number} width - pixel width at zoom level `zoomlevel`
+ * @param {Number} [zoomlevel=0] - zoomlevel at which `width` is relative to
+ * @return {Number}
+ *
+ * @example <caption>Keep feature width in meters constant with 25 pixels at zoom level 7.</caption>
+ * const s = carto.expressions;
+ * const viz = new carto.Viz({
+ *   width: s.scale(25, 7)
+ * });
+ *
+ * @example <caption>Keep feature width in meters constant with 25 pixels at zoom level 7. (String)</caption>
+ * const viz = new carto.Viz(`
+ *   width: s.scale(25, 7)
+ * `);
+ *
+ * @memberof carto.expressions
+ * @name scale
+ * @function
+ * @api
+ */
+export default class Scale extends BaseExpression {
+    constructor (width, zoomlevel = 0) {
+        width = implicitCast(width);
+        zoomlevel = implicitCast(zoomlevel);
+        super({
+            scale: div(mul(width, 0), pow(2, zoomlevel))
+        });
+        this.type = 'number';
+        this.inlineMaker = inline => inline.scale;
+    }
+    eval () {
+        return this.scale.eval();
+    }
+    _compile (metadata) {
+        checkType('scale', 'width', 0, 'number', this.scale.a.a);
+        checkType('scale', 'zoomlevel', 1, 'number', this.scale.b);
+        super._compile(metadata);
+    }
+    _preDraw (program, drawMetadata, gl) {
+        this.scale.a.b.expr = drawMetadata.scale;
```

Comment:
You need to multiply by the scale, the `0` is a placeholder that is set to the proper scale here